### PR TITLE
[RC 3.1 Release] Merging code to TX repo.

### PR DESCRIPTION
**Pull request by adityagajbhiye**
- Removed token log statement from logs.
- EDC version 0.3.0 changes for multiple BPN.
- Error handling & input validation Messages for exceptions during upload / creation.
- Manufacturer country code list.
- Cancel contract agreement on provider side.
- BPN validation in SDE.
- The field "VAN" is still case sensitive, there is no need and it causes errors if you do not know it.
- Mix of CSV-formats: When you export the data from the contract panel you get a "comma"-separated file, when you download it from history it is separated by "semicolon". Files with "comma" are shown incorrect at least on German systems.
- User not able to copy values from the contract panel as the keys are longer as the displayed field.
- To find out which contract belongs to which dataset User have to download the history file.

### CODE REVIEW

**Comment by adityagajbhiye** [comment]
@sachinargade123 @ChetanT-System Please review code and comment, so that system team can take further.

File: charts/dftbackend/templates/deployment.yaml [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#################################################################################
-# Copyright (c) 2021,2022 Catena-X
-# Copyright (c) 2021,2022 Contributors to the Eclipse Foundation
+###############################################################
+# Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
Comment (sachinargade123) [inline charts/dftbackend/templates/deployment.yaml:2]: please add 2022 in year list in copyright header

File: charts/dftbackend/templates/hpa.yaml [hunk 2]
```diff
@@ -1,3 +1,21 @@
+###############################################################
+# Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
Comment (sachinargade123) [inline charts/dftbackend/templates/hpa.yaml:2]: please add 2022 in copyright header

File: modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/processreport/model/ProcessFailureDetails.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/********************************************************************************
+ * Copyright (c) 2022 T-Systems International GmbH
```
Comment (sachinargade123) [inline modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/processreport/model/ProcessFailureDetails.java:2]: please add 2023 year in copyright header

File: InstallationGuide.md [hunk 4]
```diff
@@ -1,50 +1,39 @@
...
+For more details, please refer configuration section from [README.md](README.md)
 
+### RUN SDE backend in ArgoCD 
+ We have helm chart available for ArgoCD deployment. In deployment, if don't specified specific version, the app version/chart version is automatically picked up by ArgoCD and deployed to the environment using Helm charts.
+
+ In values.yaml you can find `default` as value for all required configuration. You need to change all those values as per your need. For reference, please refer configuration example section.
+ 
+ As part of argo CD deployment using helm chart the postgres database dependency will get provide automatic but for EDC, DigitalTwin and Portal you need to provide valid details as per configuration requirement other wise SDE service will get started with default configuration but will not work as expected.
+
+### RUN SDE Backend in k8ts cluster
```
Comment (Siegfriedk) [inline InstallationGuide.md:13]: Typo: k8s cluster

File: InstallationGuide.md [hunk 5]
```diff
@@ -1,50 +1,39 @@
...
+#### Prerequisites
+- k8ts cluster/ minikube
```
Comment (Siegfriedk) [inline InstallationGuide.md:15]: here too

File: InstallationGuide.md [hunk 6]
```diff
@@ -1,50 +1,39 @@
...
+ In values.yaml you can find `default` as value for all required configuration. You need to change all those values as per your need. For reference, please refer configuration example section.
+
+ helm repo add sde-backend https://github.com/eclipse-tractusx/dft-backend/tree/main/charts
+   
+ helm install release-name eclipse-tractusx/dft-backend
+
+### RUN SDE Backend Locally
+#### Prerequisites
+- JDK18
+- Postgres 13.2
```
Comment (Siegfriedk) [inline InstallationGuide.md:28]: No one is using postgres 13 anymore: https://app.int.demo.catena-x.net/ are you sure this is right?

File: modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/failurelog/mapper/FailureLogMapper.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+/********************************************************************************
+ * Copyright (c) 2022 T-Systems International GmbH
```
Comment (Siegfriedk) [inline modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/failurelog/mapper/FailureLogMapper.java:2]: 2023

**Review by Siegfriedk — CHANGES_REQUESTED** [review]
You are crossing the PR size limit; Please talk to your committer in your team to engage with the eclipse foundation regarding IP check: https://eclipse-tractusx.github.io/docs/release/trg-0/trg-7-03

File: charts/dftbackend/templates/deployment.yaml (hunk 1 above)
Comment (adityagajbhiye) [reply charts/dftbackend/templates/deployment.yaml]: Copy header added.

File: charts/dftbackend/templates/hpa.yaml (hunk 2 above)
Comment (adityagajbhiye) [reply charts/dftbackend/templates/hpa.yaml]: Copyright header added.

File: modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/processreport/model/ProcessFailureDetails.java (hunk 3 above)
Comment (adityagajbhiye) [reply modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/processreport/model/ProcessFailureDetails.java]: Copyright header added.

File: InstallationGuide.md (hunk 5 above)
Comment (adityagajbhiye) [reply InstallationGuide.md]: corrected corrections.

File: InstallationGuide.md (hunk 4 above)
Comment (adityagajbhiye) [reply InstallationGuide.md]: corrected corrections.

File: modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/failurelog/mapper/FailureLogMapper.java (hunk 7 above)
Comment (adityagajbhiye) [reply modules/sde-core/src/main/java/org/eclipse/tractusx/sde/core/failurelog/mapper/FailureLogMapper.java]: Copyright corrected.

File: InstallationGuide.md (hunk 6 above)
Comment (adityagajbhiye) [reply InstallationGuide.md]: Removed the postgres and marked to 11.x.x. version.

**Comment by adityagajbhiye** [comment]
Closing PR as it is too long, We will raised short PR.

**Pull request closed without merging**